File: app/src/main/java/com/example/tudoem1/MainActivity.java
package com.example.tudoem1;

import androidx.appcompat.app.AppCompatActivity;

import android.location.LocationManager;
import android.os.Bundle;
import android.widget.Button;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Build;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;
import androidx.room.Database;

import com.example.tudoem1.databaseUtils.DatabasePrototype;
import com.example.tudoem1.gpsUtils.GPService;

import util.Util;

public class MainActivity extends AppCompatActivity {

    private static final int REQUEST_CODE_BLUETOOTH_SCAN = 101;
    private DatabasePrototype db;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Util.fullscreen(this);

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
            if (ContextCompat.checkSelfPermission(this, Manifest.permission.BLUETOOTH_SCAN) != PackageManager.PERMISSION_GRANTED) {
                ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.BLUETOOTH_SCAN}, REQUEST_CODE_BLUETOOTH_SCAN);
            }
        }


        Button btnStart = findViewById(R.id.btnStart);
        Button btnSettings = findViewById(R.id.btnSettings);
        Button btnNetwork = findViewById(R.id.btnNetwork);


        btnStart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, MedicoesActivity.class);
                startActivity(intent);
                finish();
            }
        });

        btnNetwork.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if (checkGpsStatus()) {
                    Intent i = new Intent(getApplicationContext(), GPService.class);
                    startService(i);
                }

                Intent intent = new Intent(MainActivity.this, NetMonsterActivity.class);
                startActivity(intent);
                finish();
            }
        });

        btnSettings.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, SettingsActivity.class);
                startActivity(intent);
                finish();
            }
        });

    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        if (requestCode == REQUEST_CODE_BLUETOOTH_SCAN) {
            if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                // Permissão foi concedida, continue com a operação de scan
            } else {
                // Permissão foi negada, trate o caso adequadamente
            }
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        //TempDataStore temp = new TempDataStore();
    }

    @Override
    protected void onStart() {
        super.onStart();
//        db = DatabasePrototype.Companion.getDatabase(this);
//        List<MeasureStructure> measuresToUpload = db.daoNetworkMethods().getMeasuresToUpload();
//        Log.d("metrics", measuresToUpload.toString());

    }

    private boolean checkGpsStatus() {
        LocationManager locationManager = (LocationManager) this.getSystemService(LOCATION_SERVICE);
        return locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
    }
}

File: app/src/main/java/com/example/tudoem1/MainActivity.kt
package com.example.tudoem1

import android.Manifest
import android.annotation.SuppressLint
import android.content.Intent
import android.content.pm.PackageManager
import android.location.LocationManager
import android.os.Build
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import com.example.tudoem1.databaseUtils.DatabasePrototype
import com.example.tudoem1.services.GPService
import com.example.tudoem1.services.NetMonsterService
import com.example.tudoem1.services.UploadService
import util.Util

class MainActivity : AppCompatActivity() {

    private companion object {
        private const val REQUEST_CODE_BLUETOOTH_SCAN = 101
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        Util.fullscreen(this)


        if (ContextCompat.checkSelfPermission(
                this,
                Manifest.permission.BLUETOOTH_SCAN
            ) != PackageManager.PERMISSION_GRANTED
        ) {
            ActivityCompat.requestPermissions(
                this,
                arrayOf(
                    Manifest.permission.BLUETOOTH_SCAN,
                    Manifest.permission.ACCESS_COARSE_LOCATION,
                    Manifest.permission.ACCESS_FINE_LOCATION,
                    Manifest.permission.READ_PHONE_STATE
                ),
                REQUEST_CODE_BLUETOOTH_SCAN
            )

        }

        val intent1 = Intent(this@MainActivity, UploadService::class.java)
        startService(intent1)

        val btnStart: Button = findViewById(R.id.btnStart)
        val btnSettings: Button = findViewById(R.id.btnSettings)
        val btnNetwork: Button = findViewById(R.id.btnNetwork)

//        DatabasePrototype.getDatabase(this@MainActivity).clearAllTables()

        btnStart.setOnClickListener {
            startActivity(Intent(this@MainActivity, AllMeasuresActivity::class.java))
            finish()
        }

        btnNetwork.setOnClickListener {
            if (checkGpsStatus()) {
                val i = Intent(applicationContext, GPService::class.java)
                startService(i)
            }

            val i = Intent(this@MainActivity, NetMonsterService::class.java)
            startService(i)

            startActivity(Intent(this@MainActivity, AllMeasuresActivity::class.java))
            finish()
        }

        btnSettings.setOnClickListener {
            startActivity(Intent(this@MainActivity, SettingsActivity::class.java))
            finish()
        }
    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<String>,
        grantResults: IntArray
    ) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        if (requestCode == REQUEST_CODE_BLUETOOTH_SCAN) {
            if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                // Permission granted, continue with scan operation
            } else {
                // Permission denied, handle accordingly
            }
        }
    }


    private fun checkGpsStatus(): Boolean {
        val locationManager = getSystemService(LOCATION_SERVICE) as LocationManager
        return locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)
    }
}
